grafana/resource_amixr_route.go

```diff
@@ -0,0 +1,169 @@
+package grafana
+
+import (
+	"errors"
+	"log"
+	"net/http"
+
+	amixrAPI "github.com/grafana/amixr-api-go-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceAmixrRoute() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+* [Official documentation](https://grafana.com/docs/grafana-cloud/oncall/routes/)
+* [HTTP API](https://grafana.com/docs/grafana-cloud/oncall/oncall-api-reference/routes/)
+`,
+		Create: resourceAmixrRouteCreate,
+		Read:   resourceAmixrRouteRead,
+		Update: resourceAmixrRouteUpdate,
+		Delete: resourceAmixrRouteDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"integration_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the integration.",
+			},
+			"escalation_chain_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of the escalation chain.",
+			},
+			"position": &schema.Schema{
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "The position of the route (starts from 0).",
+			},
+			"routing_regex": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Python Regex query. Route is chosen for an alert if there is a match inside the alert payload.",
+			},
+			"slack": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"channel_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Slack channel id. Alerts will be directed to this channel in Slack.",
+						},
+					},
+				},
+				MaxItems:    1,
+				Description: "Slack-specific settings for a route.",
+			},
+		},
+	}
+}
+
+func resourceAmixrRouteCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*client).amixrAPI
+	if client == nil {
+		err := errors.New("amixr api client is not configured")
+		return err
+	}
```

Comment:
I have an idea to remove all of these blocks, but it's just sugar. We can wait until we migrate all of this to `...Context` functions